src/app/clusters/temperature-control-server/temperature-control-server.cpp

```diff
@@ -0,0 +1,175 @@
+/**
+ *
+ *    Copyright (c) 2023-2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ *
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/InteractionModelEngine.h>
+#include <app/util/attribute-storage.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::TemperatureControl;
+using namespace chip::app::Clusters::TemperatureControl::Attributes;
+using namespace chip::DeviceLayer;
+using chip::Protocols::InteractionModel::Status;
+
+static Status setTemperatureHandler(app::CommandHandler * commandObj, const app::ConcreteCommandPath & commandPath,
+                                    int16_t targetTemperature, uint8_t targetTemperatureLevel);
+
+namespace {
+
+class TemperatureControlAttrAccess : public AttributeAccessInterface
+{
+public:
+    // Register for the TemperatureControl cluster on all endpoints.
+    TemperatureControlAttrAccess() : AttributeAccessInterface(Optional<EndpointId>::Missing(), TemperatureControl::Id) {}
+
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+};
+
+} // namespace
+TemperatureControlAttrAccess gAttrAccess;
+
+CHIP_ERROR TemperatureControlAttrAccess::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    EndpointId endpoint = aPath.mEndpointId;
+
+    if (aPath.mClusterId != TemperatureControl::Id)
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    CHIP_ERROR status = CHIP_NO_ERROR;
+
+    switch (aPath.mAttributeId)
+    {
+    case TemperatureSetpoint::Id: {
+        int16_t tempSetpoint = 0;
+        TemperatureSetpoint::Get(endpoint, &tempSetpoint);
```

Comment:
What's the point of doing this via an AttributeAccessInterface if you're just going to read from the attribute store?  Why not just let the caler read from the attribute store itself?  This applies to all these attributes.

Reply:
Updated code